### PR TITLE
Introducing expected players instead of player count for NFR tests

### PR DESCRIPTION
**Pull request by improbable-danny**

### CODE REVIEW

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 1]
```diff
@@ -48,8 +48,8 @@ void ABenchmarkGymGameMode::GenerateTestScenarioLocations()
 	constexpr float RoamRadius = 7500.0f; // Set to half the NetCullDistance
 	{
 		FRandomStream PlayerStream;
-		PlayerStream.Initialize(FCrc::MemCrc32(&ExpectedPlayers, sizeof(ExpectedPlayers))); // Ensure we can do deterministic runs
-		for (int i = 0; i < ExpectedPlayers; i++)
+		PlayerStream.Initialize(FCrc::MemCrc32(&NumPlayers, sizeof(NumPlayers))); // Ensure we can do deterministic runs
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:51]: Why this change?

Comment (m-samiec) [reply Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:51]: Oh I thought NumPlayers previously existed

Comment (improbable-danny) [reply Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:51]: Yeah, I renamed ExpectedPlayers->NumPlayers so most of these changes are that. I updated the tests with ExpectedPlayers instead

File: Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp [hunk 2]
```diff
@@ -30,15 +30,17 @@ namespace
 
 	const FString TotalPlayerWorkerFlag = TEXT("total_players");
 	const FString TotalNPCsWorkerFlag = TEXT("total_npcs");
+	const FString ExpectedPlayersWorkerFlag = TEXT("expected_players");
 	const FString TotalPlayerCommandLineKey = TEXT("-TotalPlayers=");
 	const FString TotalNPCsCommandLineKey = TEXT("-TotalNPCs=");
+	const FString ExpectedPlayersCommandLineKey = TEXT("-ExpectedPlayers=");
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp:36]: Oh nice - I hadn't thought of doing it this way

File: Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.h [hunk 3]
```diff
@@ -22,6 +22,7 @@ class GDKTESTGYMS_API ABenchmarkGymGameModeBase : public AGameModeBase
 	static FString ReadFromCommandLineKey;
 
 	// Total number of players that will connect.
+	int32 NumPlayers;
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.h:25]: Can you update the comments here about how these variables differ

**Comment by m-samiec** [comment]
nit - when making connected PRs, please include them in the description
